theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer-contribution.ts

```diff
@@ -178,6 +186,10 @@ export class TraceViewerContribution extends WidgetOpenHandler<TraceViewerWidget
         registry.registerCommand(OpenTraceCommand, {
             execute: () => this.launchTraceServer()
         });
+        registry.registerCommand(OpenTraceWithPathCommand, {
+            isVisible: () => false,
+            execute: (path: string, options: TraceViewerWidgetOpenerOptions) => path && this.open(new URI(path), options),
+        });
```

Comment:
Please note that calling this.open(...) might also trigger launching the trace server if the server is not running (health check failure). This might be something that you don't want in your application and we might need to improve that implementation that fits the your application and the current solution with the default trace server.

Reply:
For now I do want that same flow, I think there are some discussions here to be had, but for now what is there is fine.